File: kyonggiyo-core/src/main/java/kyonggiyo/domain/auth/Platform.java
package kyonggiyo.domain.auth;

import kyonggiyo.common.exception.GlobalErrorCode;
import kyonggiyo.common.exception.InvalidValueException;

import java.util.Arrays;

public enum Platform {
    NAVER,
    KAKAO;

    public static Platform from(String platform) {
        return Arrays.stream(Platform.values())
                .filter(v -> v.name().equals(platform.toUpperCase()))
                .findAny()
                .orElseThrow(() -> new InvalidValueException(GlobalErrorCode.INVALID_REQUEST_EXCEPTION));
    }

}

File: kyonggiyo-core/src/main/java/kyonggiyo/application/auth/domain/vo/Platform.kt
package kyonggiyo.application.auth.domain.vo

import kyonggiyo.common.exception.GlobalErrorCode
import kyonggiyo.common.exception.InvalidValueException
import java.util.*

enum class Platform {
    NAVER,
    KAKAO;

    companion object {
        fun from(platform: String): Platform {
            return values().find { it.name == platform.uppercase() }
                    ?: throw InvalidValueException(GlobalErrorCode.INVALID_REQUEST_EXCEPTION)
        }
    }

}
